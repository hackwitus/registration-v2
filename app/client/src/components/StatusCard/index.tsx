import { Divider } from 'semantic-ui-react';
import * as React from 'react';

interface StatusCardProps {
  statusHeader: string;
  status: 'incomplete' | 'submitted' | 'admitted' | 'confirmed' | 'declined';
  message: string;
  registrationDeadline: Date;
  confirmationDeadline: Date;
}

const StatusCard: React.FC<StatusCardProps> = props => {
  return (
    <div className="status-card">
      <h1 className={`status-card__header status-card__header--${props.status}`}>Your Status:</h1>
      <div className={`status-card__status status-card__status--${props.status}`}>{props.status}</div>
      <Divider style={{ width: '100%' }} />
      <p className="status-card__message">{props.message}</p>
      <p className="status-card__key">
        Registration Deadline: <span className="status-card__value">{props.registrationDeadline.toUTCString()}</span>
      </p>
      <p className="status-card__key">
        Confirmation Deadline: <span className="status-card__value">{props.confirmationDeadline.toUTCString()}</span>
      </p>
    </div>
  );
};

export default StatusCard;
