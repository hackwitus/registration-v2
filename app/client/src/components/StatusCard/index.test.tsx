import * as React from 'react';
import { shallow } from 'enzyme';
import StatusCard from './index';

describe('StatusCard Component', () => {
  let wrapper;
  const mockStatusHeader = 'www.test.com';
  const mockStatus = 'confirmed';
  const mockMessage = 'Test Message';
  const mockDeadline = new Date();
  const mockConfirmation = new Date();

  wrapper = shallow(
    <StatusCard
      statusHeader={mockStatusHeader}
      status={mockStatus}
      message={mockMessage}
      registrationDeadline={mockDeadline}
      confirmationDeadline={mockConfirmation}
    />
  );

  it('renders status card', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
