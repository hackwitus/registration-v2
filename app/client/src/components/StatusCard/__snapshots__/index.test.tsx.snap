// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusCard Component renders status card 1`] = `ShallowWrapper {}`;
